tests/storage/test_event_push_actions.py

```diff
@@ -348,9 +348,12 @@ def _mark_read(event_id: str, thread_id: Optional[str] = None) -> None:
         _create_event()
         _create_event(thread_id=thread_id)
         _mark_read(event_id)
+        _assert_counts(1, 0, 3, 0)
+        _mark_read(event_id, thread_id)
         _assert_counts(1, 0, 1, 0)
 
         _mark_read(last_event_id)
+        _mark_read(last_event_id, thread_id)
```

Comment:
These tests look crazy. I am guessing a few more comments could clear up what we're trying to setup and why.

Is this really testing one scenario?

Reply:
Yes, these tests are crazy. 😢

Reply:
Can we split them up? It seems like there are many cases where we go up and back down to 0 which seems like a perfect opportunity for a new test.

At least some comments about what each little block is testing that is different from the other little blocks

Reply:
> Can we split them up? It seems like there are many cases where we go up and back down to 0 which seems like a perfect opportunity for a new test.

No, the point of the tests is that the counts stay consistent as you add new events and mark them as read.

> At least some comments about what each little block is testing that is different from the other little blocks

I plan to do this, yep.